models/xero__general_ledger.sql

```diff
@@ -1,3 +1,41 @@
+/*
+in order to generate trancking_category_name as a column we set a dictionary
+dictionary has the following appareance{'tracking_category_id' : [tuple of available id], 'tracking_category_name': [tuple of available names]}
+position match for id and name since the dictionary was generated with a sql statement 
+*/
+
+{% set tracking_category_sql %}
+    select 
+        distinct tracking_category_id,
+        regexp_replace(lower(name), r'\W', '_')   as tracking_category_name
+    from {{var('tracking_category')}}
+    where status = 'ACTIVE'
+{% endset %}
+
+
+
+{% if execute %}
+    -- fivetran_utils depends on dbt_utils 
+    {% set tc_dict = dbt_utils.get_query_results_as_dict(tracking_category_sql) %}
+    {% set tc_length = tc_dict['tracking_category_id']|length %}
+
+    -- since this case when is repetead for all source_type, we set it here as a macro and call it each time it is needed
+    {% set case_when_tracking_category %}
+        {% for n in range(0, tc_length )%}
+            case when tracking_category_id = '{{tc_dict['tracking_category_id'][n]}}'
+                -- we concate available options for that tracking_category_id 
+                then string_agg(distinct option, ' | ') 
```

Comment:
Does this mean that a single transactin could have multiple values for the same tracking category?

Reply:
Yes, exactly! To match the BS, we were using 'Multiple categories' when that happened, but we talked with Chago and we decided to concatenate the different values.

Reply:
Interesting, good to know for reporting purrposes